### DISCUSSION TITLE
Getting blank image on exporting map as PNG

### DISCUSSION THREAD

**Discussion by SushantKumar29** [visgl/deck.gl#6909] [Q&A]
I am exporting the map as a PNG file, but after downloading the map, the image is blank.

`

    function handleDownloadMap() {
        const fileName = 'Map.png';
        if (!mapRef.current || !deckRef?.current) {
            return;
        }
        const mapGL = mapRef?.current.getMap();
        const deck = deckRef?.current.deck;

        const mapboxCanvas = mapGL.getCanvas();
        deck.redraw(true);
        const deckglCanvas = deck.canvas;

        let merge = document.createElement('canvas');
        merge.width = mapboxCanvas.width;
        merge.height = mapboxCanvas.height;

        var context = merge.getContext('2d');

        context.globalAlpha = 1.0;
        context?.drawImage(mapboxCanvas, 0, 0);
        context.globalAlpha = 1.0;
        context?.drawImage(deckglCanvas, 0, 0);

        merge.toBlob((blob: any) => {
            FileSaver.saveAs(blob, fileName);
        });
    };
`

`

    <DeckGL
        ref={deckRef}
        layers={layers}
        effects={lightingEffects}
        viewState={viewState}
    >
        <StaticMap
            mapStyle={MAP_STYLE}
            preventStyleDiffing={true}
            mapboxApiAccessToken={mapboxKey}
            ref={mapRef}
            gl={glContext}
            onLoad={onMapLoad}
         />
        <Button
            onClick={handleDownloadMap}
        >
            Download Map
        </Button>
    </DeckGL>
`
Please help

**Comment by Pessimistress** [comment]
https://github.com/visgl/deck.gl/issues/4436#issuecomment-610472868

**Reply by SushantKumar29** [reply]
Can you tell me what changes I need to make in my code ? I debugged the download function and got that the ViewState of the deck is 'null'. Can that be the issue? If yes can you tell me the fix?

**Reply by radoslavirha** [reply]
@SushantKumar29 
just add: 
```
new Deck({
  ...
  glOptions: {
    preserveDrawingBuffer: true
  },
  ...
})
```

**Reply by j-oca** [reply]
I am having the same problem.

@SushantKumar29 this didn't solve the problem, right? You are using React and, in theory, you don't need the DrawingBuffer option. Moreover, that is not available through the React component.

Did you find a solution?

**Reply by radoslavirha** [reply]
@j-oca try `deck.redraw(true);` again? Sometimes I got blank image too, but I solved it with `deck.redraw(true);` if I remember
